src/services/tradeApiClient.types.js

```diff
@@ -1167,35 +1170,44 @@ function providerItemTransform(providerItem, providerType) {
   // transformedResponse.dailyReturns = transformedResponse.dailyReturns.sort(
   //   (a, b) => a.name.getTime() - b.name.getTime(),
   // );
-
-  if (!transformedResponse.isCopyTrading) {
-    // Updating followers count because it's out of date for clones
-    transformedResponse.followers = transformedResponse.aggregateFollowers.length
-      ? transformedResponse.aggregateFollowers[transformedResponse.aggregateFollowers.length - 1]
-          .totalFollowers
-      : 0;
-    transformedResponse.newFollowers = calculateNewFollowers(transformedResponse);
-  }
-  let connectedOnly = providerType.startsWith("connected");
-  let copyTraders = false;
+  let connectedOnly = providerType ? providerType.startsWith("connected") : false;
+  let copyTrader = false;
   let profitSharingProvider = false;
+  let CTorPS = false;
 
+  // This first check is for legacy api
   if (!providerType) {
-    copyTraders = transformedResponse.provType === "copytrading";
+    copyTrader = transformedResponse.provType === "copytrading";
     profitSharingProvider = transformedResponse.provType === "profitsharing";
   } else if (!connectedOnly) {
-    copyTraders = providerType === "copy_trading";
+    copyTrader = providerType === "copy_trading";
     profitSharingProvider = providerType === "profit_sharing";
+    CTorPS = !["signal_providers", "connected_providers"].includes(providerType);
   } else {
-    copyTraders = providerType === "connected_traders" && !transformedResponse.profitSharing;
+    copyTrader = providerType === "connected_traders" && !transformedResponse.profitSharing;
+    CTorPS = providerType === "connected_traders";
     profitSharingProvider =
       providerType === "connected_traders" && transformedResponse.profitSharing;
   }
 
   transformedResponse.providerLink = `/${
-    copyTraders ? "copyTraders" : profitSharingProvider ? "profitSharing" : "signalProviders"
+    copyTrader ? "copyTraders" : profitSharingProvider ? "profitSharing" : "signalProviders"
   }/${transformedResponse.id}`;
 
+  if (!copyTrader && !profitSharingProvider) {
+    // Updating followers count because it's out of date for clones
+    transformedResponse.followers = transformedResponse.aggregateFollowers.length
+      ? transformedResponse.aggregateFollowers[transformedResponse.aggregateFollowers.length - 1]
+          .totalFollowers
+      : 0;
+    transformedResponse.newFollowers = calculateNewFollowers(transformedResponse);
+  }
+
+  transformedResponse.profitSharing = profitSharingProvider;
+  transformedResponse.copyTrader = copyTrader;
+  transformedResponse.CTorPS = CTorPS;
```

Comment:
I'm wondering if it'd be cleaner/better to keep provType and remove profitSharing, copyTrader, CTorPS. But let's keep it for now.

Reply:
I thought of the same but then again, that was redundant, this way we know ourself on FE, but you are right, It would have been cleaner. And to be honest I dont understand why we need to remove those properties because I dont think there will be any performance improvement.